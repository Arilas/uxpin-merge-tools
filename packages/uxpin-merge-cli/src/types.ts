export { ComponentImplementationInfo } from './steps/discovery/component/ComponentInfo';
export { ImplSerializationResult } from './steps/serialization/component/implementation/ImplSerializationResult';
export { serializeJSComponent } from './steps/serialization/component/implementation/javascript/serializeJSComponent';
export { serializeTSComponent } from './steps/serialization/component/implementation/typescript/serializeTSComponent';
export { DesignSystemSnapshot } from './steps/serialization/DesignSystemSnapshot';
