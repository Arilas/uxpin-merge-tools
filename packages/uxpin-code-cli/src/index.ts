export { getDesignSystemMetadata } from './getDesignSystemMetadata';
export { getDesignSystemSummary } from './getDesignSystemSummary';
