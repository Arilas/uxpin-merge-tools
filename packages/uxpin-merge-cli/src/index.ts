import 'source-map-support/register';

export { runProgram } from './program/runProgram';
export { Command } from './program/command/Command';
export { getToolVersion } from './program/utils/version/getToolVersion';

export { ComponentImplementationInfo } from './steps/discovery/component/ComponentInfo';
export { ImplSerializationResult } from './steps/serialization/component/implementation/ImplSerializationResult';
export { serializeJSComponent } from './steps/serialization/component/implementation/javascript/serializeJSComponent';
export { serializeTSComponent } from './steps/serialization/component/implementation/typescript/serializeTSComponent';
export { DesignSystemSnapshot } from './steps/serialization/DesignSystemSnapshot';
