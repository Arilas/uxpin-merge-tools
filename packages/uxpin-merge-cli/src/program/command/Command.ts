export enum Command {
  DUMP = 'dump',
  EXPERIMENT = 'experiment',
  INIT = 'init',
  CREATE_APP = 'create-app',
  GENERATE_PRESETS = 'generate-presets',
  PUSH = 'push',
  SERVER = 'server',
  SUMMARY = 'summary',
}

export const DEFAULT_COMMAND:Command = Command.EXPERIMENT;
