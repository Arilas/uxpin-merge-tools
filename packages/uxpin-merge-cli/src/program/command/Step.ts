import { DSMetadata } from '../DSMeta';

export interface Step {
  shouldRun:boolean;
  exec:(infos:DSMetadata) => any;
}
