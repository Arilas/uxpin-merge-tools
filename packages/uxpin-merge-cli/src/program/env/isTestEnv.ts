import { Environment } from './Environment';

export function isTestEnv():boolean {
  return process.env.NODE_ENV === Environment.TEST;
}
