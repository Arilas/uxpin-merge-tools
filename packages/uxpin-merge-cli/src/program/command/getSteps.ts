import { BuildOptions } from '../../steps/building/BuildOptions';
import { ProgramArgs, PushProgramArgs, ServerProgramArgs } from '../ProgramArgs';
import { getDumpCommandSteps } from './dump/getDumpCommandSteps';
import { getPushCommandSteps } from './push/getPushCommandSteps';
import { getServerCommandSteps } from './server/getServerCommandSteps';
import { Step } from './Step';
import { getSummaryCommandSteps } from './summary/getSummaryCommandSteps';

export function getSteps(args:ProgramArgs):Step[] {
  if (args.command === 'server') {
    return getServerCommandSteps(getBuildOptions(args), args);
  }

  if (args.command === 'push') {
    return getPushCommandSteps(getBuildOptions(args));
  }

  if (args.command === 'dump') {
    return getDumpCommandSteps();
  }

  if (args.command === 'summary') {
    return getSummaryCommandSteps();
  }

  return [];
}

function getBuildOptions(args:ServerProgramArgs | PushProgramArgs):BuildOptions {
  const { cwd, webpackConfig, wrapper } = args;
  return {
    projectRoot: cwd,
    webpackConfigPath: webpackConfig,
    wrapperPath: wrapper,
  };
}
