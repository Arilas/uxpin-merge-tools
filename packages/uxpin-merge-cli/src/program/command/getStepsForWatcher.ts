import { ProgramArgs } from '../args/ProgramArgs';
import { getExperimentationWatchCommandSteps } from './experimentation/getExperimentationWatchCommandSteps';
import { Step } from './Step';

export function getStepsForWatcher(args:ProgramArgs):Step[] {
  if (args.command === 'experiment') {
    return getExperimentationWatchCommandSteps(args);
  }

  return [];
}
