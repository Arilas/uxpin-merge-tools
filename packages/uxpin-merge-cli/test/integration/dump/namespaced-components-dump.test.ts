import { Command } from '../../../src';
import { Environment } from '../../../src/program/env/Environment';
import { emptyLatestCommitStub } from '../../resources/stubs/emptyLatestCommit';
import { runUXPinMergeCommand } from '../../utils/command/runUXPinMergeCommand';
import { setTimeoutBeforeAll } from '../../utils/command/setTimeoutBeforeAll';
import { setupStubbyServer } from '../../utils/stubby/setupStubbyServer';

const CURRENT_TIMEOUT:number = 60000;
setTimeoutBeforeAll(CURRENT_TIMEOUT);

jest.mock('../../../src/program/utils/version/getToolVersion');

describe('The dump command', () => {
  const { getTlsPort } = setupStubbyServer(emptyLatestCommitStub);

  describe('run for the namespacedComponents repository', () => {
    it('prints the JSON describing the full repository', async () => {
      // when
      const output:string = await runUXPinMergeCommand({
        cwd: 'resources/designSystems/namespacedComponents',
        env: {
          UXPIN_API_DOMAIN: `0.0.0.0:${getTlsPort()}`,
          UXPIN_ENV: Environment.TEST,
        },
        params: [
          Command.DUMP,
          '--webpack-config "./webpack.config.js"',
        ],
      });

      // then
      expect(JSON.parse(output)).toMatchSnapshot();
    });
  });
});
