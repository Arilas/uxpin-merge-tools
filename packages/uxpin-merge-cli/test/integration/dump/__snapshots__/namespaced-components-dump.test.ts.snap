// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The dump command run for the namespacedComponents repository prints the JSON describing the full repository 1`] = `
Object {
  "categorizedComponents": Array [
    Object {
      "components": Array [
        Object {
          "documentation": Object {
            "examples": Array [],
          },
          "info": Object {
            "dirPath": "src/components/Card",
            "implementation": Object {
              "framework": "reactjs",
              "lang": "typescript",
              "path": "src/components/Card/Card.tsx",
            },
            "presets": Array [],
          },
          "name": "Card",
          "presets": Array [],
          "properties": Array [
            Object {
              "description": "",
              "isRequired": true,
              "name": "children",
              "type": Object {
                "name": "node",
                "structure": Object {},
              },
            },
          ],
        },
        Object {
          "documentation": Object {
            "examples": Array [],
          },
          "info": Object {
            "dirPath": "src/components/Card/components/Header",
            "implementation": Object {
              "framework": "reactjs",
              "lang": "typescript",
              "path": "src/components/Card/components/Header/Header.tsx",
            },
            "presets": Array [],
          },
          "name": "Header",
          "namespace": Object {
            "importSlug": "Card_Header",
            "name": "Card",
          },
          "presets": Array [],
          "properties": Array [
            Object {
              "description": "",
              "isRequired": true,
              "name": "children",
              "type": Object {
                "name": "node",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": true,
              "name": "title",
              "type": Object {
                "name": "string",
                "structure": Object {},
              },
            },
          ],
        },
        Object {
          "documentation": Object {
            "examples": Array [],
          },
          "info": Object {
            "dirPath": "src/components/Card/components/Header/components/Button",
            "implementation": Object {
              "framework": "reactjs",
              "lang": "typescript",
              "path": "src/components/Card/components/Header/components/Button/Button.tsx",
            },
            "presets": Array [],
          },
          "name": "Button",
          "namespace": Object {
            "importSlug": "Card_Header_Button",
            "name": "Card.Header",
          },
          "presets": Array [],
          "properties": Array [
            Object {
              "description": "",
              "isRequired": true,
              "name": "children",
              "type": Object {
                "name": "node",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "isPrimary",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
          ],
        },
      ],
      "name": "Card",
    },
  ],
  "name": "Example Design System",
  "vcs": ObjectContaining {
    "branchName": Any<String>,
    "commitHash": StringMatching /\\[a-z0-9\\]\\+/,
  },
}
`;
